bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/ErrorUtils.java

```diff
@@ -133,4 +144,40 @@ public static BError createUnorderedTypesError(Object lhsValue, Object rhsValue)
         throw createError(BallerinaErrorReasons.UNORDERED_TYPES_ERROR, BLangExceptionHelper.getErrorMessage(
                 RuntimeErrors.UNORDERED_TYPES_IN_COMPARISON, lhsValue, rhsValue));
     }
+
+    public static BError createRuntimeError(Module module, RuntimeErrorType errorType, Object... params) {
+        return createRuntimeError(module, errorType, getErrorDetail(getErrorMessage(errorType, params)), null);
+    }
+
+    public static BError createRuntimeError(Module module, RuntimeErrorType errorType,
+                                            BMap<BString, Object> detail, BError cause) {
+        try {
+            return createError(module, errorType.getErrorName(), getModulePrefixedErrorName(module, errorType),
+                    cause, detail);
+        } catch (BError e) {
+            throw ErrorCreator.createError(StringUtils.fromString(e.getMessage()));
+        }
+    }
+
+    public static BString getErrorMessage(RuntimeErrorType runtimeErrors, Object... params) {
+        return StringUtils.fromString(MessageFormat
+                .format(messageBundle.getString(runtimeErrors.getErrorMsgKey()), params));
+    }
+
+    public static BMap<BString, Object> getErrorDetail(BString errMessage) {
```

Comment:
Shall we make this method private?

Reply:
Yes, we can make it as private for now. But it might be needed to access from outside when we migrating all other runtime errors with the new error creator api.